src/sentry/api/bases/organization.py

```diff
@@ -280,6 +282,21 @@ class FilterParams(TypedDict, total=False):
     environment_objects: list[Environment] | None
 
 
+def _validate_fetched_projects(
+    filtered_projects: Sequence[Project],
+    slugs: set[str] | None,
+    ids: set[int] | None,
+) -> None:
+    """
+    Validates that user has access to the specific projects they are requesting.
+    """
+    missing_project_ids = ids and ids != {p.id for p in filtered_projects}
+    missing_project_slugs = slugs and slugs != {p.slug for p in filtered_projects}
+
+    if missing_project_ids or missing_project_slugs:
+        raise PermissionDenied
```

Comment:
nit [performance]: if we know there are missing project ids, we can avoid doing the next check by returning early. If you want to keep the lines of code for readability, you could make them lambda functions and then this way we will avoid doing the second function/logic if the first returns a value